lib/echo_rpc/echo_rpc.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/**
+ * @packageDocumentation
+ * @module echo_rpc
+ * @mergeTarget
+ */
+
+import * as model from "./model"
+import * as model_utils from "./model_utils";
+import * as eventstream_rpc from "../eventstream_rpc";
+
+export {model};
+
+export class Client {
+    private rpcClient : eventstream_rpc.RpcClient;
+    private serviceModel : eventstream_rpc.EventstreamRpcServiceModel;
+
+    constructor(config: eventstream_rpc.RpcClientConfig) {
+        this.serviceModel = model_utils.makeServiceModel();
+        this.rpcClient = eventstream_rpc.RpcClient.new(config);
+    }
+
+    async connect() : Promise<void> {
+        await this.rpcClient.connect();
+    }
+
+    close() : void {
+        this.rpcClient.close();
+    }
+
+    async echoMessage(request : model.EchoMessageRequest, options?: eventstream_rpc.OperationOptions) : Promise<model.EchoMessageResponse> {
+        return new Promise<model.EchoMessageResponse>(async (resolve, reject) => {
+            try {
+                let operationConfig : eventstream_rpc.OperationConfig = {
+                    name: "awstest#EchoMessage",
+                    client: this.rpcClient,
+                    options: (options) ? options : {}
+                };
+
+                let operation : eventstream_rpc.RequestResponseOperation<model.EchoMessageRequest, model.EchoMessageResponse> =
+                    new eventstream_rpc.RequestResponseOperation<model.EchoMessageRequest, model.EchoMessageResponse>(operationConfig, this.serviceModel);
+
+                let response : model.EchoMessageResponse = await operation.activate(request);
+
+                resolve(response);
+            } catch (err) {
+                reject(err);
+            }
+        });
```

Comment:
I don't think you need the "new Promise" stuff at all. 

just:
```
                let operationConfig : eventstream_rpc.OperationConfig = {
                    name: "awstest#EchoMessage",
                    client: this.rpcClient,
                    options: (options) ? options : {}
                };

                let operation : eventstream_rpc.RequestResponseOperation<model.EchoMessageRequest, model.EchoMessageResponse> =
                    new eventstream_rpc.RequestResponseOperation<model.EchoMessageRequest, model.EchoMessageResponse>(operationConfig, this.serviceModel);

               return await operation.activate(request);
```

Reply:
using `async` makes the function return a promise already, you don't need to make your own.

Reply:
I'm not sure that exceptions thrown outside the explicit await would manifest as rejecting the implicit promise.  Would need to experiment

Reply:
they do

Reply:
appears to work, still feels uncomfortable

Reply:
At the very least, it seems reasonable to use this simplification on the generated client's request response promises due to their simplicity.  Not sure I want to mess with the internal promises yet though.

Reply:
:)
This is the point of `async`, you don't need to do raw promises anymore. "It just works" ™️

Reply:
Looking further, a lot of the internal async functions do a kind of "calculus change of variables" where they await on promises of one type (raw eventstream messages) but resolve to another type (modeled responses), so we may just limit this to the service client.